File: Aula 1 - Revisão de POO (Desafio)/Filme.py
'''
    - file: Filme.py
    - Author: Diego Anestor Coutinho (diego.anestor@gec.inatel.br)
'''

class Filme:
    def __init__(self, titulo, ano, genero, duracao):
        self.titulo = titulo
        self.ano = ano
        self.genero = genero
        self.duracao = duracao
        self.assistido = False
        self.avaliacao = None

    def info(self):
        print(f"Título: {self.titulo}")
        print(f"Ano: {self.ano}")
        print(f"Gênero: {self.genero}")
        print(f"Duração: {self.duracao} minutos")

        print("Assistido: ", end="")
        if self.assistido:
            print("Sim")
        else:
            print("Não")
            
        if self.avaliacao:
            print(f"Avaliação: {self.avaliacao}/10")

    def assistido(self):
        self.assistido = True

    def avaliar(self, nota):
        self.avaliacao = nota

File: film_list/Filme.js
/*
    - file: Filme.js
    - Author: Diego Anestor Coutinho (diego.anestor@gec.inatel.br)
*/

class Filme {
    constructor(titulo, ano, genero, duracao) {
        this.titulo = titulo;
        this.ano = ano;
        this.genero = genero;
        this.duracao = duracao;
        this.assistido = false;
        this.avaliacao = null;
    }

    info() {
        console.log(`Título: ${this.titulo}`);
        console.log(`Ano: ${this.ano}`);
        console.log(`Gênero: ${this.genero}`);
        console.log(`Duração: ${this.duracao} minutos`);

        process.stdout.write("Assistido: ");
        if (this.assistido) {
            console.log("Sim");
        } else {
            console.log("Não");
        }

        if (this.avaliacao) {
            console.log(`Avaliação: ${this.avaliacao}/100`);
        }
    }

    assistir() {
        this.assistido = true;
    }

    avaliar(nota) {
        this.avaliacao = nota;
    }
}

module.exports = Filme;